docs/hello_nextflow/05_hello_channels.md

```diff
@@ -0,0 +1,558 @@
+
+So to recap, we're going to develop a workflow that does the following:
+
+_ADD joint genotyping FLOWCHART_
+
+<figure class="excalidraw">
+--8<-- "docs/hello_nextflow/img/haplotype-caller.excalidraw.svg"
+</figure>
+
+1. Generate an index file for each BAM input file using Samtools
+2. Run the GATK HaplotypeCaller on each BAM input file to generate a GVCF of per-sample genomic variant calls
+3. Collect all the GVCFs and combine them into a data store
+4. Run joint genotyping on the combined GVCF data store to produce a cohort-level VCF
+
+### Dataset
+
+-   **A reference genome** consisting of a small region of the human chromosome 20 (from hg19/b37) and its accessory files (index and sequence dictionary).
+-   **Three whole genome sequencing samples** corresponding to a family trio (mother, father and son), which have been subset to a small portion on chromosome 20 to keep the file sizes small. The sequencing data is in [BAM](https://samtools.github.io/hts-specs/SAMv1.pdf) (Binary Alignment Map) format, i.e. genome sequencing reads that have already been mapped to the reference genome.
+-   **A list of genomic intervals**, i.e. coordinates on the genome where our samples have data suitable for calling variants, provided in BED format.
+
+---
+
+## 0. Warmup: Run Samtools and GATK directly
+
+Just like previously, we want to try out the commands manually before we attempt to wrap them in a workflow.
+
+!!! note
+
+     Make sure you're in the correct working directory:
+     `cd /workspace/gitpod/hello-nextflow`
+
+### 0.1. Index a BAM input file with Samtools
+
+This first step is the same as in Part 2: Hello-GATK so you can skip it if you've already done that in this session.
+
+#### 0.1.1. Pull the samtools container
+
+```bash
+docker pull community.wave.seqera.io/library/samtools:1.20--b5dfbd93de237464
+```
+
+#### 0.1.2. Spin up the container interactively
+
+```bash
+docker run -it -v ./data:/data community.wave.seqera.io/library/samtools:1.20--b5dfbd93de237464
+```
+
+#### 0.1.3. Run the indexing command
+
+```bash
+samtools index /data/bam/reads_mother.bam
+```
+
+#### 0.1.4. Check that the BAM index has been produced
+
+```bash
+ls /data/bam/
+```
+
+This should show:
+
+```console title="Output"
+reads_father.bam      reads_mother.bam      reads_mother.bam.bai  reads_son.bam
+```
+
+Where `reads_mother.bam.bai` has been created as an index to `reads_mother.bam`.
+
+#### 0.1.5. Exit the container
+
+```bash
+exit
+```
+
+### 0.2. Call variants with GATK HaplotypeCaller in GVCF mode
+
+This second step is **different** from Part 2: Hello-GATK, since now we are running GGATK in 'GVCF mode', so you **should NOT skip it**.
```

Comment:
```suggestion
This second step is **different** from Part 2: Hello-GATK, since now we are running GATK in 'GVCF mode', so you **should NOT skip it**.
```